projects/components/src/metric-card/metric-card.component.ts

```diff
@@ -1,33 +1,48 @@
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
 import { Color } from '@hypertrace/common';
 import { IconSize } from '../icon/icon-size';
+import { MetricCardLayout } from '../public-api';
 import { MetricCardIndicatorType } from './metric-card';
 @Component({
   selector: 'ht-metric-card',
   styleUrls: ['./metric-card.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="metric-card" [style.backgroundColor]="this.backgroundColor" [style.borderColor]="this.borderColor">
-      <div class="indicator-and-title">
-        <div class="indicator">
-          <ng-container [ngSwitch]="this.indicator">
-            <ng-container *ngSwitchCase="'${MetricCardIndicatorType.Dot}'">
-              <div class="dot" [style.backgroundColor]="this.indicatorColor"></div>
+    <div
+      class="metric-card"
+      [ngClass]="{ clickable: this.clickable }"
+      [ngStyle]="this.getStylesForCard()"
+      (click)="this.onSelectionChange()"
+    >
+      <div class="card-with-indicator" *ngIf="this.layoutType === '${MetricCardLayout.CardWithIndicator}'">
```

Comment:
If you find yourself ngIf'ing with two different templates and the two modes have different inputs, they shouldn't be the same component. Create a different component for the use case.

Reply:
we have different layouts for almost the same set of inputs, so won't it be better to keep these layouts in the same component instead of creating multiple small components with same inputs?

Reply:
Even if they're identical, what do we save by having two different implementations inside the same component? The boilerplate of declaring a new component? It's the same reason that putting a flag argument on a method with otherwise identical inputs to affect its behavior is an antipattern. If it's a small styling difference and the majority of the template is reused that's one thing (e.g. primary vs secondary button), but if it's a completely different template that just makes each change that much harder - you have to track each input and output to two different places in code.

Anyway it's up to you guys, just my 2 cents.

Reply:
Actually it's even more problematic, because on top of the layout type there's already an indicator type, and now the two don't play nice. The new layout type ignores the indicator type, and it uses inputs differently. If I'm using an icon indicator, it gets its colors from indicatorColor; if I'm using cardWithLargeIcon, it gets its color from iconColor.

So at a minimum, please make your change by extending the existing inputs so everything behaves consistently.

Reply:
Creating a new component for it